File: cpp-sdk/src/objects/output/DocTheme.cpp
#include "DocTheme.h"

DocTheme::DocTheme() {}
DocTheme::~DocTheme() {}

void DocTheme::Serialize(Json::Value& root) {
    root["title"] = title;
    root["sentiment_score"] = sentiment_score;
    root["evidence"] = evidence;
    root["is_about"] = is_about;
    root["strength_score"] = strength_score;
}

void DocTheme::Deserialize(Json::Value& root) {
    title = root.get("title", "").asString();
    sentiment_score = root.get("sentiment_score", 0.0).asDouble();
    evidence = root.get("evidence", 0).asUInt();
    is_about = root.get("is_about", false).asBool();
    strength_score = root.get("strength_score", 0.0).asDouble();
}

File: java-sdk/com/semantria/objects/output/DocTheme.java
package com.semantria.objects.output;
import javax.xml.bind.annotation.*;

public final class DocTheme
{
	private String title;
	private Float sentiment_score;
	private Integer evidence;
	private Boolean is_about;
	private Float strength_score;
	
	@XmlElement(name="title")
	public String getTitle() { return title; }
	@XmlElement(name="sentiment_score")
	public Float getSentimentScore() { return sentiment_score; }
	@XmlElement(name="evidence")
	public Integer getEvidence() { return evidence; }
	@XmlElement(name="is_about")
	public Boolean getIsAbout() { return is_about; }
	@XmlElement(name="strength_score")
	public Float getStrengthScore() { return strength_score; }
	
	public void setTitle(String ttitle) { title = ttitle; }
	public void setSentimentScore(Float tsentiment) { sentiment_score = tsentiment; }
	public void setEvidence(Integer tevidence) { evidence = tevidence; }
	public void setIsAbout(Boolean about) { is_about = about; }
	public void setStrengthScore(Float tscore) { strength_score = tscore; }
}
